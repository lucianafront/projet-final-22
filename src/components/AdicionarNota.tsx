import * as React from "react";
import { NotasContext } from "../context/notas.context";
import { NotasContextType } from "../@types/notas.d";

import { FaPlus } from "react-icons/fa6";

const AdicionarNota = () => {
  const { nota, criandoNota, atualizarNota, salvarNota } = React.useContext(
    NotasContext
  ) as NotasContextType;

  const handleInputChange = (
    e: React.ChangeEvent<
      HTMLInputElement | HTMLTextAreaElement | HTMLSelectElement
    >
  ) => {
    criandoNota({
      ...nota,
      [e.target.name]: e.target.value,
    });
  };

  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    if (nota.id !== 0) {
      atualizarNota(nota.id, nota);
    } else {
      salvarNota(nota);
    }
  };

  return (

        <form onSubmit={handleSubmit}>
          <input
            type="text"
            required
            name="titulo"
            placeholder="Titulo"
            value={nota.titulo}
            onChange={handleInputChange}
          />

          <textarea
            name="conteudo"
            rows={5}
            required
            placeholder="Conteudo"
            value={nota.conteudo}
            onChange={handleInputChange}
          ></textarea>
            <label>Prioridade 

   
          <select id="prioridade"
            name="prioridade"
            value={nota.prioridade}
            onChange={handleInputChange}
            >
            <option value="baixa">Baixa</option>
            <option value="media">Media</option>
            <option value="alta">Alta</option>
          </select>
            </label>
          <button title="Adicionar nota" type="submit">
            <FaPlus />
          </button>
        </form>

  );
};

export default AdicionarNota;
