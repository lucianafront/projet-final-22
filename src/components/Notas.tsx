import * as React from "react";
import { NotasContext } from "../context/notas.context";
import { INotas, NotasContextType } from "../@types/notas.d";

import { FaTrashCan } from "react-icons/fa6";
import { FaRotate } from "react-icons/fa6";

const Notas = () => {
  
  const {
    editandoNota,
    deletarNota,
    notas,
    notasFiltradas,
    setNotasFiltradas,
  } = React.useContext(NotasContext) as NotasContextType;

  const filtrarNotas = (prioridade: string) => {
    console.log(prioridade);
    setNotasFiltradas(
      notas.filter((nota: INotas) => {
        if (prioridade === "") {
          return notas;
        } else {
          return nota.prioridade === prioridade;
        }
      })
    );
  };

  return (
    <div>
      <div className="circulos">
        <div className="circle todas" title="Mostrar todas as notas" onClick={() => filtrarNotas("")}></div>
        <div title="Mostar somente notas com prioridade baixa"
          className="circle baixa"
          onClick={() => filtrarNotas("baixa")}
        ></div>
        <div title="Mostar somente notas com prioridade media"
          className="circle media"
          onClick={() => filtrarNotas("media")}
        ></div>
        <div 
        title="Mostar somente notas com prioridade alta"
        className="circle alta" onClick={() => filtrarNotas("alta")}></div>
      </div>

      <ul className="note">
        {notasFiltradas.map((nota: INotas) => (
          <li key={nota.id} className={nota.prioridade} data-testid={`nota-id-${nota.id}`} >
            <h2>{nota.titulo}</h2>
            <p>{nota.conteudo}</p>

            <p style={{ paddingTop: "50px", fontSize: "0.8rem" }}>
              {nota.data.toLocaleDateString()} -- {nota.prioridade}
            </p>
            <button title="Remover nota"  onClick={() => deletarNota(nota.id)} className="delete">
              <FaTrashCan className="icones" />
            </button>
            <button title="Atualizar nota" onClick={() => editandoNota(nota.id)} className="atualiza">
              <FaRotate className="icones" />
            </button>
          </li>
        ))}
      </ul>
    </div>
  );
};

export default Notas;
