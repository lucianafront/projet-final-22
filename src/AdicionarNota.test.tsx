import React, {act} from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import AdicionarNota from "./components/AdicionarNota";
import NotasProvider from "./context/notas.context";
import Notas from "./components/Notas";

describe("AdicionarNota component", () => {

    test("matches snapshot", () => {
    const { container } =     render(<AdicionarNota />,{ wrapper: NotasProvider})
    expect(container).toMatchSnapshot();
});
  
  
    test("renders AdicionarNota component", () => {
      render(<AdicionarNota />,{ wrapper: NotasProvider})


    const tituloInput = screen.getByPlaceholderText("Titulo");
    expect(tituloInput).toBeInTheDocument();

    const conteudoTextarea = screen.getByPlaceholderText("Conteudo");
    expect(conteudoTextarea).toBeInTheDocument();

    const prioridadeSelect = screen.getByLabelText("Prioridade");
    expect(prioridadeSelect).toBeInTheDocument();

    const adicionarButton = screen.getByTitle("Adicionar nota");
    expect(adicionarButton).toBeInTheDocument();
  });


  test("adiciona uma nova nota", () => {
    render(<AdicionarNota />,{ wrapper: NotasProvider})

    const tituloInput = screen.getByPlaceholderText("Titulo");
    fireEvent.change(tituloInput, { target: { value: "Nota 4" } });
    expect(tituloInput).toHaveValue("Nota 4");

    const conteudoTextarea = screen.getByPlaceholderText("Conteudo");
    fireEvent.change(conteudoTextarea, { target: { value: "Conteudo 4" } });
    expect(conteudoTextarea).toHaveValue("Conteudo 4");

    const prioridadeSelect = screen.getByLabelText("Prioridade");
    fireEvent.change(prioridadeSelect, { target: { value: "alta" } });
    expect(prioridadeSelect).toHaveValue("alta");

    const adicionarButton = screen.getByTitle("Adicionar nota");
    fireEvent.click(adicionarButton);

    const novo = screen.getByPlaceholderText("Titulo");
    expect(novo).toHaveValue("");
  });



});