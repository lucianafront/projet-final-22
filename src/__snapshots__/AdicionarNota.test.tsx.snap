// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdicionarNota component adiciona uma nova nota 1`] = `undefined`;

exports[`AdicionarNota component matches snapshot 1`] = `
<div>
  <form>
    <input
      name="titulo"
      placeholder="Titulo"
      required=""
      type="text"
      value=""
    />
    <textarea
      name="conteudo"
      placeholder="Conteudo"
      required=""
      rows="5"
    />
    <label>
      Prioridade
      <select
        id="prioridade"
        name="prioridade"
      >
        <option
          value="baixa"
        >
          Baixa
        </option>
        <option
          value="media"
        >
          Media
        </option>
        <option
          value="alta"
        >
          Alta
        </option>
      </select>
    </label>
    <button
      title="Adicionar nota"
      type="submit"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32V224H48c-17.7 0-32 14.3-32 32s14.3 32 32 32H192V432c0 17.7 14.3 32 32 32s32-14.3 32-32V288H400c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z"
        />
      </svg>
    </button>
  </form>
</div>
`;
